algorithms/unit_tests/TestRandom.hpp

```diff
@@ -469,6 +488,276 @@ struct TestDynRankView {
     ASSERT_LE(val.max_val, max);
   }
 };
+
+template <class Pool, class ViewType>
+struct TestView_Normal {
+  using ExecutionSpace = typename Pool::device_type::execution_space;
+  using ScalarA        = typename ViewType::value_type;
+  ViewType A;
+  const int n;
+  const ScalarA mean;
+  const ScalarA stddev;
+
+  TestView_Normal(ViewType A_, int n_, ScalarA mean_, ScalarA stddev_)
+      : A(A_), n(n_), mean(mean_), stddev(stddev_) {}
+
+  KOKKOS_FUNCTION void operator()(int i,
+                                  NormalRandomProperties<ScalarA>& prop) const {
+    for (int k = 0; k < static_cast<int>(A.size()); k += n) {
+      ScalarA tmp = *(A.data() + i + k);
+      prop.mean += tmp;
+      prop.variance += (tmp - mean) * (tmp - mean);
+    }
+  }
+
+  void run() {
+    uint64_t ticks =
+        std::chrono::high_resolution_clock::now().time_since_epoch().count();
+
+    Pool random(ticks);
+    ExecutionSpace exec;
+    Kokkos::fill_random_normal(A, random, mean, stddev);
+
+    NormalRandomProperties<ScalarA> val;
+    Kokkos::parallel_reduce(Kokkos::RangePolicy<ExecutionSpace>(exec, 0, n),
+                            *this, val);
+
+    exec.fence();
+
+    ScalarA mean_calc = val.mean / A.size();
+    ScalarA var_calc  = val.variance / (A.size() - 1);
+
+    ScalarA mean_eps     = Kokkos::abs(mean - mean_calc);
+    ScalarA variance_eps = Kokkos::abs((stddev * stddev) / var_calc - 1.0);
+    std::cout << "Rank " << ViewType::rank << ": mean_eps " << mean_eps
+              << ", variance_eps " << variance_eps << ", mean_expect " << mean
+              << ", variance_expect " << stddev * stddev << ", mean "
+              << mean_calc << ", variance " << var_calc << std::endl;
+    if (std::is_same<ScalarA, double>::value) {
+      EXPECT_LE(mean_eps, 1e-2);
+      EXPECT_LE(variance_eps, 1e-2);
+    } else {
+      EXPECT_LE(mean_eps, 1e-1);
+      EXPECT_LE(variance_eps, 1e-1);
+    }
```

Comment:
How did you come up with these tolerances? What's the probability for this test to pass given the runtime length?

Reply:
@masterleinad, When testing on Lassen, this test always pass with the tolerance of 1e-3 on cuda, openmp, serial backends with both float and double. However, it almost fails right away with CI testings on CPUs. So, I chose 1e-2 for double and 1e-1 for float just for the worst test cases pass. Do you have any suggestion?

Reply:
The tolerances should depend on the sample size, right? The distribution of the sample mean is a normal distribution with the same mean but with a standard deviation scaled by `1./sqrt(n)` and we should use that in the assertion (since we are using different sample sizes on the host and on the device).